src/poetry/console/commands/self/source/add.py

```diff
@@ -0,0 +1,19 @@
+from __future__ import annotations
+
+from poetry.console.commands.self.self_command import SelfCommand
+from poetry.console.commands.source.add import SourceAddCommand
+
+
+class SelfSourceAddCommand(SelfCommand, SourceAddCommand):
```

Comment:
I followed the pattern established in `src/poetry/console/commands/self/add.py`.